ArduinoFreevalve/ArduinoFreevalve.ino

```diff
@@ -0,0 +1,84 @@
+// An Arduino program for controlling valve timing on a 6.5HP Predator engine from Harbor Freight.
+//
+// License goes here?
+//
+// Wesley Kagan, 2020
+// Ric Allinson, 2020
+
+// Pins assignment.
+const int HALL_MAGNET = 2;
+const int EXHAUST_V = 12;
+const int INTAKE_V = 13;
+
+// Control constants.
+const int DEG_PER_MAGNET = 6; // Number of degrees for per magnet.
+const int ROTATION_TWO = 760/DEG_PER_MAGNET;     // Number of interrupts in the full cycle.
+const int ROTATION_ONE = (760/DEG_PER_MAGNET)-1; // Number of interrupts in a half cycle.
+
+// Runtime mapping variables.
+bool intakeMap[ROTATION_TWO] = {};  // Intake open/close mapping is equal the total number of interrupts for two rotations.
+bool exhaustMap[ROTATION_TWO] = {}; // Exhaust open/close mapping is equal the total number of interrupts for two rotations.
+
+// ISR variables.
+volatile int hallCounter = 0;           // The number of magnets after the last TDC.
+volatile bool secondRotation = false;   // "true" if the cam is on its second rotation.
+volatile bool printLog = false;         // Used to stop duplicate values from printing in the loop.
+volatile unsigned long timeGap = 0;     // Function level time between interrupts.
+volatile unsigned long lastTimeGap = 0; // Global level time between interrupts.
+
+void setup() {
+  Serial.begin(115200);
+  pinMode(HALL_MAGNET, INPUT);
+  attachInterrupt(0, magnetDetect, RISING);
+  pinMode(INTAKE_V, OUTPUT);
+  pinMode(EXHAUST_V, OUTPUT);
+  // Load mappings here.
+  // intakeMap = ?
+  // exhaustMap = ?
+}
+
+void loop() {
+  // This may not print all values. Only the most recent from the last interrupt.
+  if(printLog){
+    Serial.print(lastTimeGap);
+    Serial.print(hallCounter);
+    Serial.print(secondRotation);
+    printLog = false;
+  }
+}
+
+// This function will be called about every 15.79ms at the maximum RPM of 3800.
```

Comment:
This interrupt happens much faster than 15.79ms.  Spaced at 6 degrees over a full rotation:

(1 rev / 60 pulses) * (1 min / 3800 rev) * (60 sec / min) = 1 / 3800 sec/pulse = ~263.16usec.  The missing tooth will come by roughly every 15.79 ms.

Reply:
Again with my maths! I've updated the comments with your correct maths. Thanks.